src/OPLS.js

```diff
@@ -358,57 +415,64 @@ export class OPLS {
       Eh.sub(tOrth.mmul(pOrth));
       // prediction
       tPred = Eh.mmul(model.plsC.w.transpose());
-      yHat = tPred.mmul(model.plsC.betas).mmul(model.plsC.q);
+      const components = tPred
+        .mmul(model.plsC.betas)
+        .mmul(model.plsC.q.transpose());
+      totalPred = new Matrix(components.rows, 1);
+      for (let i = 0; i < components.rows; i++) {
+        totalPred.setRow(i, [components.getRowVector(i).sum()]);
+      }
     }
 
     if (labels?.rows > 0) {
       if (this.mode === 'regression') {
         const tssy = tss(labels);
-        const press = tss(labels.clone().sub(yHat));
+        const press = tss(labels.clone().sub(totalPred));
```

Comment:
Would it be possible in general to use longer variable name to help to understand the code ?

Reply:
Would be nice! I don't know what @jwist think about it?